### PR TITLE
[#2091]: [Subtask] Standardize the invocation method for the Builder pattern

### PR DESCRIPTION
**Pull request by hiirrxnn**
What changes were proposed in this pull request?
Standardize the invocation method for the Builder pattern

Why are the changes needed?
To standardize the invocation method for the Builder pattern

Fix: #2091 

Does this PR introduce any user-facing change?
N/A

How was this patch tested?
N/A

### CODE REVIEW

File: common/src/main/java/com/datastrato/gravitino/dto/CatalogDTO.java
Comment (mchades) [inline common/src/main/java/com/datastrato/gravitino/dto/CatalogDTO.java:1]: The default constructor should be changed from `public` to `private` too

File: common/src/main/java/com/datastrato/gravitino/dto/MetalakeDTO.java
Comment (mchades) [inline common/src/main/java/com/datastrato/gravitino/dto/MetalakeDTO.java:1]: The default constructor should be changed from public to private too

**Comment by mchades** [comment]
The warning is caused by Java 17. You can reproduce it locally by modifying `jdkVersion = 8` to `jdkVersion = 17` in the `gradle.properties` and then running the build command.

**Comment by mchades** [comment]
Overall, LGTM. Please resolve the conflicts.

**Comment by mchades** [comment]
Hi @hiirrxnn , Thank you for your hard work, I saw this PR has been blocked for a long time, did you still work on this PR? or do you need any support?

**Comment by charliecheng630** [comment]
@hiirrxnn @mchades If you need any support, I can address this issue.

**Comment by mchades** [comment]
> @hiirrxnn @mchades If you need any support, I can address this issue.

Maybe I think you can rebase this PR and move on with the work. @charliecheng630  

How do you think? @hiirrxnn

**Comment by hiirrxnn** [comment]
All the required work has been done and approved by other contributors . We just need to make sure CI passes. One file has been causing some issues

**Comment by jerryshao** [comment]
I think @charliecheng630 already fixed them, thanks @hiirrxnn for your contribution, I'm going to close this.

**Pull request closed without merging**